### PR TITLE
Autorest violations fixes in RecoveryServices.Backup

### PR DESCRIPTION
**Pull request by DheerendraRathor**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [x] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by DheerendraRathor** [comment]
@alvadb Please review this.

**Comment by alvadb** [comment]
Working on this.  Hope to have it done by tomorrow.

File: specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2016-12-01/bms.json [hunk 1]
```diff
@@ -3969,179 +5298,164 @@
...
-    "ProtectedItem": {
-      "description": "Base class for backup items.",
+    "PreBackupValidation": {
+      "description": "Pre-backup validation for Azure VM Workload provider.",
       "type": "object",
       "properties": {
-        "protectedItemType": {
-          "description": "backup item type.",
+        "status": {
+          "description": "Status of protectable item, i.e. InProgress,Succeeded,Failed",
```
Comment (alvadb) [inline specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2016-12-01/bms.json]: Should this be an enum?

File: specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2017-07-01/bms.json [hunk 2]
```diff
@@ -0,0 +1,3568 @@
...
+        },
+        "resourceId": {
+          "type": "string"
+        },
+        "poLogicalName": {
+          "type": "string"
+        }
+      }
+    },
+    "BackupStatusResponse": {
```
Comment (alvadb) [inline specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2017-07-01/bms.json]: Please add description to properties.

File: specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2017-07-01/bms.json [hunk 3]
```diff
@@ -0,0 +1,3568 @@
...
+          "description": "The list of backup management usages for the given vault.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/BackupManagementUsage"
+          }
+        }
+      }
+    },
+    "BackupStatusRequest": {
+      "description": "BackupStatus request.",
```
Comment (alvadb) [inline specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2017-07-01/bms.json]: Please add description to properties.

File: specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2017-07-01/bms.json [hunk 4]
```diff
@@ -0,0 +1,3568 @@
...
+    },
+    "Job": {
+      "description": "Defines workload agnostic properties for a job.",
+      "required": [
+        "jobType"
+      ],
+      "type": "object",
+      "properties": {
+        "entityFriendlyName": {
+          "descrip
```
Comment (alvadb) [inline specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2017-07-01/bms.json]: Please add description to properties.

Comment (alvadb) [inline specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2017-07-01/bms.json]: Please add description to properties.

Comment (alvadb) [inline specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2017-07-01/bms.json]: Please add description.

**Review by alvadb — CHANGES_REQUESTED** [review]

File: specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2016-12-01/bms.json [hunk 5]
```diff
@@ -2102,200 +2606,793 @@
...
+          "description": "Additional details of a workload container."
+        }
+      }
+    },
+    "AzureWorkloadContainerExtendedInfo": {
+      "description": "Extended information of the container.",
+      "type": "object",
+      "properties": {
+        "hostServerName": {
+          "description": "Host Os Name in case of Stand Alone and \r\n            Cluster Name in case of distributed container.",
```
[the remark this answers is not in the document]
Comment (DheerendraRathor) [reply specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2016-12-01/bms.json]: Why is this necessary?

File: specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2017-07-01/bms.json (hunk 4 above)
Comment (DheerendraRathor) [reply specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2017-07-01/bms.json]: Description is already present in this case.

File: specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2017-07-01/bms.json (hunk 3 above)
Comment (DheerendraRathor) [reply specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2017-07-01/bms.json]: For all missing descriptions for properties, we will take them in next iteration as this iteration is targeted for validation errors mainly for ARM compliance.

File: specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2017-07-01/bms.json (hunk 2 above)
Comment (DheerendraRathor) [reply specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2017-07-01/bms.json]: Done!

File: specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2016-12-01/bms.json [hunk 6]
```diff
@@ -2102,200 +2606,793 @@
...
+          }
+        },
+        "healthDetails": {
+          "description": "Health details on this backup item.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/AzureIaaSVMHealthDetails"
+          }
+        },
+        "lastBackupStatus": {
```
[the remark this answers is not in the document]
Comment (DheerendraRathor) [reply specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2016-12-01/bms.json]: No. I've removed misleading description.

**Comment by DheerendraRathor** [comment]
@alvadb Updated PR.

**Comment by jianghaolu** [comment]
@alvadb this PR looks good to me. Please feel free to merge if all your feedbacks are addressed.

**Review by alvadb — APPROVED** [review]

**Pull request merged**